src/dev/impl/DevToys/ViewModels/Tools/EncodersDecoders/Base64EncoderDecoder/Base64EncoderDecoderToolViewModel.cs

```diff
@@ -200,6 +200,11 @@ private async Task<string> DecodeBase64DataAsync(string? data)
                 return string.Empty;
             }
 
+            if (data![data.Length - 1] != '=')
```

Comment:
This will add a "=" padding to the end of `data` even if it is already valid. By always padding an "=" to the end of a value without one, it will fail to decode valid user input.
I'd suggest checking using something like this `data.Length % 4 != 0` to check if data should be valid first
More info, this SE answer is quite indepth: [https://stackoverflow.com/questions/4080988/why-does-base64-encoding-require-padding-if-the-input-length-is-not-divisible-by/18518605#18518605](https://stackoverflow.com/questions/4080988/why-does-base64-encoding-require-padding-if-the-input-length-is-not-divisible-by/18518605#18518605)

Reply:
@Regenhardt , hello, did you get a chance to look at this? :)

Reply:
Hi, yep had a look and an idea, just not enough time to actually do it right now. Exams currently and I have way too much on my plate until like, middle of next week. Will definitely push something soon!

Reply:
Sounds good, thank you for the heads up. There's no rush. Take your time. Good luck with your exams :)

Reply:
Hello :) Any update by any chance?

Reply:
Yup sorry for the delay, made it add variable padding now depending on how close to the 4 chars it goes.

Reply:
Hello,
That's awesome! thank you very much for the update :D And no worries for the delay!
I will go ahead and merge this PR.